design/gateway-apis-implementation.md

```diff
@@ -163,23 +169,26 @@ The output of this watcher is Envoy configuration.
 ## Detailed Design
 
 ### Configuration
-Contour will have an entry added to the config file for the Gateway it should watch.
+Contour will have an entry added to the config file for the `ControllerName` it should watch.
+The `ControllerName` takes the format of `projectcontour.io/<namespace>/contour` which requires the `<namespace>` param to be specified matching the namespace where Contour is deployed. 
 
 This will look like this:
 ```yaml
 gateway:
-  name: gatewayname
-  namespace: gatewaynamespace
+  controllerName: 
+  name: gatewayname  # <--- deprecate
+  namespace: gatewaynamespace  # <-- deprecate
 ```
 
-The current flag `--experimental-service-apis` will be removed.
-The boolean it represents will be deduced from the following rules:
-- the gateway is configured in the config file.
-- the referenced gateway exists.
+Contour will look for a Gateway + GatewayClass that matches the `ControllerName` field specified in the configuration file.
```

Comment:
This probably needs to be broken up to be more specific, so that we say that Contour will find a single valid GatewayClass, and then any Gateways that reference the GatewayClass.

Similarly, below should say that if Contour doesn't find any matching GatewayClasses, then no Gateways will be selected until there is one.

Reply:
Yup it states that right below that sentence:
```
If it does not find a matching Gateway/GatewayClass, then it will begin processing GatewayAPI resources once they are available. 
```

Reply:
oh, 🤦 . Thanks.